### PR TITLE
Java: Automodel, new candidates fix

### PR DESCRIPTION
**Pull request by kaeluka**
Following some discussions (backref), this updates a few details of the newly extracted automodel candidates and examples.

 - adds `extensibleType` to metadata
 - extracts method doc and class doc as in framework mode. However, these will often not be available, as in application mode, the method we build a model will generally live in a different code base. I suggest coming back to this topic once we craft prompts.

### CODE REVIEW

File: java/ql/automodel/test/AutomodelApplicationModeExtraction/AutomodelApplicationModeExtractCandidates.expected [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+| PluginImpl.java:5:27:5:37 | name | command-injection, path-injection, request-forgery, sql-injection\nrelated locations: $@, $@, $@.\nmetadata: $@, $@, $@, $@, $@, $@, $@, $@, $@, $@. | PluginImpl.java:5:27:5:37 | name | CallContext | PluginImpl.java:5:27:5:37 | name | MethodDoc | PluginImpl.java:5:27:5:37 | name | ClassDoc | file://hudson:1:1:1:1 | hudson | package | file://Plugin:1:1:1:1 | Plugin | type | file://true:1:1:1:1 | true | subtypes | file://configure:1:1:1:1 | configure | name | file://(String,String):1:1:1:1 | (String,String) | signature | file://:1:1:1:1 |  | input | file://Parameter[0]:1:1:1:1 | Parameter[0] | output | file://false:1:1:1:1 | false | isVarargsArray | file://:1:1:1:1 |  | alreadyAiModeled | file://sourceModel:1:1:1:1 | sourceModel | extensibleType |
```
Comment (tausbn) [inline java/ql/automodel/test/AutomodelApplicationModeExtraction/AutomodelApplicationModeExtractCandidates.expected:1]: I'm a bit bothered by the fact that we use the location of the endpoint in the case where there is no related location to report (though I realise that this has been part of the design for a long time). This makes it difficult to easily tell apart which related locations are "real" and which are not. I wonder if we could get away with using an [option type](https://github.com/github/codeql/blob/49d57653dc5e739531dea535f0d6593de800bdca/shared/util/codeql/util/Option.qll) somehow. Or perhaps represent "no related location" as some default placeholder value.

Also, at a glance I couldn't see any of the tests actually populating any of the two newly added related locations (though I may have missed something -- which again points to this output not being terribly readable).

**Review by tausbn — APPROVED** [review]
One comment (which I am fine with leaving as future work), otherwise I think this looks good (and hence give my approval now). 👍

File: java/ql/automodel/test/AutomodelApplicationModeExtraction/AutomodelApplicationModeExtractCandidates.expected (hunk 1 above)
Comment (kaeluka) [reply java/ql/automodel/test/AutomodelApplicationModeExtraction/AutomodelApplicationModeExtractCandidates.expected:1]: I agree; also, an option type may actually be a good solution here. There's no tests, indeed. I'll add one!

Comment (kaeluka) [reply java/ql/automodel/test/AutomodelApplicationModeExtraction/AutomodelApplicationModeExtractCandidates.expected]: [928898e](https://github.com/github/codeql/pull/14184/commits/928898e9794149e1044c8bf22c9fb35c1999d3e4)

**Review by tausbn — APPROVED** [review]
Great! 👍

**Pull request merged**